src/modules/fancyzones/lib/ZoneSet.cpp

```diff
@@ -433,23 +473,28 @@ void ZoneSet::CalculateGridLayout(Rect workArea, JSONHelpers::ZoneSetLayoutType
             }
         }
     }
-    CalculateGridZones(workArea, gridLayoutInfo, spacing);
+    return CalculateGridZones(workArea, gridLayoutInfo, spacing);
 }
 
-void ZoneSet::CalculateUniquePriorityGridLayout(Rect workArea, int zoneCount, int spacing) noexcept
+bool ZoneSet::CalculateUniquePriorityGridLayout(Rect workArea, int zoneCount, int spacing) noexcept
 {
-    CalculateGridZones(workArea, predefinedPriorityGridLayouts[zoneCount - 1], spacing);
+    if (zoneCount <= 0 || zoneCount >= sizeof(predefinedPriorityGridLayouts))
+    {
+        return false;
+    }
+
+    return CalculateGridZones(workArea, predefinedPriorityGridLayouts[zoneCount - 1], spacing);
 }
 
-void ZoneSet::CalculateCustomLayout(Rect workArea, const std::wstring& customZoneSetFilePath, int spacing) noexcept
+bool ZoneSet::CalculateCustomLayout(Rect workArea, const std::wstring& customZoneSetFilePath, int spacing) noexcept
 {
     wil::unique_cotaskmem_string guuidStr;
     if (SUCCEEDED_LOG(StringFromCLSID(m_config.Id, &guuidStr)))
     {
         JSONHelpers::FancyZonesDataInstance().GetCustomZoneSetFromTmpFile(customZoneSetFilePath, guuidStr.get());
         const auto& zoneSet = JSONHelpers::FancyZonesDataInstance().GetCustomZoneSetsMap()[guuidStr.get()];
 
-        if (zoneSet.type == JSONHelpers::CustomLayoutType::Canvas)
+        if (zoneSet.type == JSONHelpers::CustomLayoutType::Canvas && std::holds_alternative<JSONHelpers::CanvasLayoutInfo>(zoneSet.info))
```

Comment:
nit: `std::holds_alternative` could be checked above and return false immediately